### ISSUE TITLE
App hangs and circles on Re-Authentication Needed dialog

### ISSUE THREAD

**Issue by ohporter** [vector-im/element-android#6490]
### Steps to reproduce

1. Where are you starting? What can you see?

Start app on phone. See the Encryption upgrade available banner at the top.

![screenshot1](https://user-images.githubusercontent.com/650260/177594264-32e3fd60-1e37-426b-8618-5cf45ac456be.png)

2. What do you click?

Tap on the Encryption upgrade available banner. The Re-Authentication Needed box appears and circles endlessly. Swiping back will bring up the Skip/Continue dialog. Skip will clear out the authentication dialog. Continue will cause it to continue circling.

3. Killing and restarting the app brings up the Encryption upgrade available banner again.

#### What did you expect?

I expected to be able to authenticate and not have this banner come up again.

#### What happened instead?

Circles forever at Re-Authentication Needed

![Screenshot_20220706-114800](https://user-images.githubusercontent.com/650260/177594124-c19909ee-0a8e-4c83-ab5e-0aeb103e68d9.png)
d

### Your phone model

Google Pixel 5

### Operating system version

Android 12

### Application version and app store

Element Version 1.4.26 [40104262] (G-b9381) olm version 3.2.12 Google Play

### Homeserver

non-public Synapse 1.61.1

### Will you send logs?

Yes

### Are you willing to provide a PR?

Yes

**Comment by ouchadam** [comment]
is it possible to swipe away the banner and continue to use the app, or has the app become non functional?

**Comment by ohporter** [comment]
> is it possible to swipe away the banner and continue to use the app, or has the app become non functional?

Yes, it's possible to swipe it away and continue using the app. It reappears whenever the app is started.

**Comment by HanEmile** [comment]
I've got a person with a case in which it is not possible to swipe it away.

**Comment by lub** [comment]
As a workaround it's possible to bootstrap cross-signing via settings -> security -> cross-signing or on a different client. Then the popup also disappeared.

**Comment by davidegirardi** [comment]
Another workaround is to login on Element Web/Desktop to trigger the cross-sign. This happened to friend of mine who registered an account on matrix.org directly from the mobile app on the first start.

**Comment by gschintgen** [comment]
Hi, I'm completely new to Matrix and Element.

For me it's the same as fir @davidegirardi 's friend: I signed up using Element yesterday (only to browse through the settings and such) and now on the first 'real' use of the app I get the popup leading to the endless circling. If I hit Android's back button, a warning pops up (text can't be selected for copy pasting): if you cancel now you may lose encrypted messages & data (...)

That's a rather disconcerting first contact I'm having with Matrix... Hopefully this is reproducible (and hence fixable).

**Comment by GRAYgoose124** [comment]
I don't know all the details, but my friend signed up to matrix directly through the app on his android device and is hitting this exact case.

**Comment by Alan-R** [comment]
Same issue.

**Comment by HeIIow2** [comment]
happened to a friend of mine, who registered there account on matrix.org from the app as well, so there might be something more to it.

**Comment by HeIIow2** [comment]
> Another workaround is to login on Element Web/Desktop to trigger the cross-sign. This happened to friend of mine who registered an account on matrix.org directly from the mobile app on the first start.

happened to a friend of mine, who registered there account on matrix.org from the app as well, so there might be something more to it.

**Comment by trankten** [comment]
Hello. This issue happens when SSO / OAuth login are setup and the account is logged using a 3rd party system. There's a loop because the client is not able to contact the login provider. As @lub said, doing it manually will fix it.

**Comment by weeman1337** [comment]
This happened yesterday annoyingly with the Android session being the only session. Logs sent.

**Comment by pabpas** [comment]
Recently onboarded a small group of people to Element and all the ones installing from google play store are having this problem. The ones on iOS or using the Fdroid version are not affected.

**Comment by jaywink** [comment]
Just onboarding some people and some got hit by this. It's a Synapse v1.83.0 server using SSO.

**Comment by languagemaniac** [comment]
> Recently onboarded a small group of people to Element and all the ones installing from google play store are having this problem. The ones on iOS or using the Fdroid version are not affected.

Uninstalled it and installed it from F-Droid and I keep having this issue.

**Comment by firefoxlover** [comment]
I had this issue too today, not on matrix.org

The solution really is to log in in the web interface, cancel the dialog in the app and verify the session on the app using the dialog above. I dont know why the browser is trusted though.

This happened on SchildiChat and on Element (F-Droid), not on Element iOS.

**Comment by calvinrw** [comment]
One of my users encountered this bug on a Samsung Galaxy S8 running Android 9 / One UI 1.0, with Element 1.6.5 from Google Play. This is a new user on my homeserver, whose only session is this Android phone.

I had the user follow @davidegirardi 's advice and attempt to bootstrap the cross-signing process through Element's Settings. Element complained with "authentication failed" when he entered his password, though he swears he entered it correctly (and I believe him), so it didn't seem like it completed. Interestingly, the green banner stopped showing up after that.

I had him submit a rageshake with description "duplicate of 6490" as its description a little before I posted this.

**Comment by mfat** [comment]
This issue is very annoying. I've installed element on all phones of family members, now that we are far from each other I'm not able to call any of them.

Met one of them in person, skipped reauthentication on their device and only after that they can receice calls.

**Comment by canadaduane** [comment]
I have this issue on mobile as well. However, when I try to log into the web app and cross-sign as a workaround, it reports "Unable to query secret storage status", then quits the settings page.

Using Firefox, connecting to matrix.vmst.io.
